frontend/src/viewer/components/2d/path-element.ts

```diff
@@ -319,6 +319,35 @@ export class PathElement extends connect(store)(LitElement) {
     return `http://xctrack.org/xcplanner?${params}`;
   }
 
+  private encodeNumber(num: number): string {
+    // google.maps.geometry.encoding.encodePath is for latLng only.
+    let pnum = num << 1;
+    if (num < 0) {
+      pnum = ~pnum;
+    }
+    let result = '';
+    while (pnum > 0x1f) {
+      result += String.fromCharCode(((pnum & 0x1f) | 0x20) + 63);
+      pnum = pnum >>> 5;
+    }
+    result += String.fromCharCode(63 + pnum);
+    return result;
+  }
+
+  private getXctsk(): string {
```

Comment:
Could this return the object instead and move `XCTSK:${JSON.stringify(obj)}` to the call site ?

Reply:
This is the required format, I see no point in not returning it.

Reply:
Making it easier to test.
At some point I also want to save/load xctsk files

Reply:
Done.